ruby_programming/testing_with_rspec/test_driven_development.md

```diff
@@ -0,0 +1,37 @@
+# Test-Driven Development
+
+## Introduction
+
+Hopefully, you've been having fun developing in Ruby thus far. Perhaps one thing you may **not** have enjoyed as much, though, is having to run your script on the command line, running through each step manually, only to find that it doesn't return what you expected. And then comes the *even-less-fun* part: You tug at your hair a bit, change a few variables, and run your script again through each step, **hoping** that the results are correct.
+
+Folks, we're here to tell you there *IS* better. Hoping is for that exam you didn't study for and took by the seat of your pants. **Knowing** is test-driven development.
+
+## Learning Outcomes
+
+- Understand what test-driven development (TDD) is, and why it's important.
```

Comment:
Would adding "What are the three stages of a TDD cycle?" be a good additional learning outcome?

Reply:
Excellent idea :+1: